### PR TITLE
v1.10 backports 2021-12-06

### PR DESCRIPTION
**Pull request by nathanjsweet**
* #18031 -- ci: Restart pods when toggling KPR switch (@brb)
 * #18018 -- test/contrib: Bump CoreDNS version to 1.8.3 (@brb)
 * #18048 -- bpf: Add WireGuard to complexity and compile tests (@pchaigno)
 * #18068 -- docs: KUBECONFIG for cilium-cli with k3s (@kkourt)
 * #18006 -- test: Fix incorrect selector for netperf-service (@christarazi)
 * #18104 -- test: Extend coredns clusterrole with additional resource permissions (@aditighag)
 * #18113 -- service: Always allocate higher ID for svc/backend (@brb)
 * #18110 -- docs: fix eksctl ClusterConfig to allow copy (@aanm)
 * #18123 -- docs: fix link to signoff / certificate of origin section (@timoreimann)

Once this PR is merged, you can update the PR labels via:
```upstream-prs
$ for pr in 18031 18018 18048 18068 18006 18104 18113 18110 18123; do contrib/backporting/set-labels.py $pr done 1.10; done
```
or with
```
$ make add-label branch=v1.10 issues=18031,18018,18048,18068,18006,18104,18113,18110,18123
```

### CODE REVIEW

**Comment by joestringer** [comment]
/test-backport-1.10

**Review by christarazi — APPROVED** [review]

File: bpf/complexity-tests/netnext/bpf_lxc.txt [hunk 1]
```diff
@@ -1,3 +1,4 @@
...
+-DSKIP_DEBUG=1 -DENABLE_IPV6=1 -DENABLE_HOST_SERVICES_TCP=1 -DENABLE_HOST_SERVICES_UDP=1 -DENABLE_HOST_REDIRECT=1 -DENABLE_ROUTING=1 -DNO_REDIRECT=1 -DPOLICY_VERDICT_NOTIFY=1 -DALLOW_ICMP_FRAG_NEEDED=1 -DENABLE_IDENTITY_MARK=1 -DMONITOR_AGGREGATION=3 -DCT_REPORT_FLAGS=0x0002 -DENABLE_HOST_FIREWALL=1 -DHAVE_LPM_TRIE_MAP_TYPE=1 -DHAVE_LRU_HASH_MAP_TYPE=1 -DENABLE_MASQUERADE=1 -DENABLE_SRC_RANGE_CHECK=1 -DENABLE_NODEPORT=1 -DENABLE_EXTERNAL_IP=1 -DENABLE_NODEPORT_ACCELERATION=1 -DENABLE_SESSION_AFFINITY=1 -DENABLE_DSR_ICMP_ERRORS=1 -DENABLE_DSR=1 -DENABLE_HOSTPORT=1 -DENABLE_LOADBALANCER -DENABLE_DSR_HYBRID=1 -DENABLE_IPV4_FRAGMENTS=1 -DENABLE_TPROXY=1 -DENABLE_REDIRECT_FAST=1 -DENABLE_BANDWIDTH_MANAGER=1 -DENABLE_CUSTOM_CALLS=1 -DETH_HLEN=0 -DENCAP_IFINDEX=1 -DTUNNEL_MODE=1 -DENABLE_EGRESS_GATEWAY=1 -DENABLE_WIREGUARD
+
```
Comment (pchaigno) [inline bpf/complexity-tests/netnext/bpf_lxc.txt]: Unnecessary newline here.

**Review by pchaigno — CHANGES_REQUESTED** [review]
One (probably) nit below. I'm not exactly sure how K8sVerifier will react to that newline so it could be breaking.

**Comment by nathanjsweet** [comment]
/test-backport-1.10

**Review by pchaigno — APPROVED** [review]
Looks good for my PR :pray:

**Review by aditighag — APPROVED** [review]
Backport of my commit looks good. Thanks!

**Review by brb — APPROVED** [review]
My commits LGTM, thanks.

**Review by aanm — APPROVED** [review]

**Comment by nbusseneau** [comment]
Travis hit an unspecified infra issue with the ARM build, retriggering.

**Comment by nbusseneau** [comment]
AWS-CNI and EKS workflows failed due to AWS-side infrastructure transient issues, retriggering.

**Comment by nbusseneau** [comment]
/test-1.16-netnext

**Comment by nbusseneau** [comment]
It looks like the L4LB workflow might have hit an actual issue, see logs at https://github.com/cilium/cilium/actions/runs/1550474705/attempts/1. I'm retriggering to confirm.

**Review by kkourt — APPROVED** [review]

**Comment by nbusseneau** [comment]
/ci-l4lb-1.10

**Comment by nbusseneau** [comment]
/ci-awscni-1.10

**Comment by nbusseneau** [comment]
/ci-eks-1.10

**Comment by joestringer** [comment]
I ran the ci-l4lb-1.10 workflow on PR #18210 which has effectively no changes vs. the plain v1.10 branch, and it hit the same failure (#18211). We should investigate this, but given that this backport does not introduce the failure, there is no reason to believe that this is a regression introduced by these changes. 

[EKS run](https://github.com/cilium/cilium/runs/4473736147?check_suite_focus=true) hit https://github.com/cilium/cilium/issues/16938 which is now quarantined on master.

[net-next run](https://jenkins.cilium.io/job/Cilium-PR-K8s-1.16-net-next/2162/testReport/junit/Suite-k8s-1/16/K8sFQDNTest_Validate_that_FQDN_policy_continues_to_work_after_being_updated/) hit an issue with the newly introduced test, investigating.

**Pull request merged**